src/components/ChatComponent/Sections/ChatQueueSection.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+interface IProps {
+  queueMembers: string[];
+  createRoomWith;
+}
+
+const ChatQueueSection = (props: IProps) => {
+  const createList = () => {
```

Comment:
Det er litt overhead ved å lage en funksjon og kjøre den for hver render (som skjer her). En mer konvensjonell måte å gjøre det på er f.eks. slik, og så bytte ut `createList()` på linje 22 med `queue`:
```tsx
const queue = useMemo(
    () => props.queueMembers.map((student, index) => (
        <li key={index}>
          {student}
          <button onClick={() => props.createRoomWith(student)}>Chat</button>
        </li>
    )),
    [props.queueMembers]
);
```

`useMemo` vil lagre resultatet av funksjonen den første gangen komponenten rendres, og vil ikke kalkulere resultatet på nytt før ev. `props.queueMembers` endres.